### DISCUSSION TITLE
What is the impact of SELinux on CRI-O

### DISCUSSION THREAD

**Discussion by alpana17** [cri-o/cri-o#7991] [General]
There are multiple modes in which SELinux can be set. What is the impact of each mode on CRI-O?
Also, what is the idle setting for SELinux for CRI-O?
Different modes of SELinux are:
1. Enabled with PERMISSIVE mode
2. Enabled with ENFORCING mode
3. Disabled

According to https://www.redhat.com/en/blog/latest-container-exploit-runc-can-be-blocked-selinux document:
SELinux should be enabled in CRI-O to prevent certain vulnerabilities

**Comment by alpana17** [comment]
@haircommander please see

**Reply by haircommander** [reply]
cri-o has an option `selinux` which is a boolean that allows a user to enable selinux separation for the container's storage. it only works if the node is enabled. We recommend setting it and setting the node to ENFORCING

**Reply by alpana17** [reply]
@haircommander I see from https://github.com/cri-o/cri-o/blob/main/docs/crio.conf.5.md that selinux option is deprecated.
Does it read from host only??

**Reply by alpana17** [reply]
Also, shall I take it as selinux in enforcing mode on host machine?

**Reply by haircommander** [reply]
yeah it's deprecated because soon we will interpret just the selinux enforcing mode on the host. So yeah set to enforcing on the host machine and `selinux = true` in cri-o and you should be good

**Reply by alpana17** [reply]
@haircommander selinux=true in /etc/crio/crio.conf
Right?

**Reply by haircommander** [reply]
yup correct!

**Reply by alpana17** [reply]
Thank you
